ClientRegressionWithMockNetworkTest.*

This is a failure from my local runs, so no CloudBees link is available.

```
java.lang.AssertionError
    at org.junit.Assert.fail(Assert.java:86)
    at org.junit.Assert.assertTrue(Assert.java:41)
    at org.junit.Assert.assertTrue(Assert.java:52)
    at com.hazelcast.client.ClientRegressionWithMockNetworkTest.testLock_WhenClientAndOwnerNodeDiesTogether(ClientRegressionWithMockNetworkTest.java:582)
    at com.hazelcast.client.ClientRegressionWithMockNetworkTest.testLock_WhenDummyClientAndOwnerNodeDiesTogether(ClientRegressionWithMockNetworkTest.java:563)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:497)
    at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
    at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
    at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
    at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
    at com.hazelcast.test.AbstractHazelcastClassRunner$ThreadDumpAwareRunAfters.evaluate(AbstractHazelcastClassRunner.java:132)
    at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
    at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
    at com.hazelcast.test.HazelcastParallelClassRunner.access$301(HazelcastParallelClassRunner.java:46)
    at com.hazelcast.test.HazelcastParallelClassRunner$TestRunner.run(HazelcastParallelClassRunner.java:139)
    at java.lang.Thread.run(Thread.java:745)
```

https://gist.github.com/Donnerbart/aaed8a5ed47ea92d327f

The failing assert is this one:

```
assertTrue(lock.tryLock(1, TimeUnit.MINUTES));
```

**ClientRegressionWithMockNetworkTest.testClientConnectionEvents**

```
java.lang.AssertionError: null
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at com.hazelcast.client.ClientRegressionWithMockNetworkTest.testClientConnectionEvents(ClientRegressionWithMockNetworkTest.java:315)
```

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x-IbmJDK1.7/com.hazelcast$hazelcast-client/810/testReport/junit/com.hazelcast.client/ClientRegressionWithMockNetworkTest/testClientConnectionEvents/

```
java.lang.AssertionError: null
	at com.hazelcast.client.spi.impl.ClientInvocationServiceSupport.shutdown(ClientInvocationServiceSupport.java:173)
	at com.hazelcast.client.spi.impl.ClientSmartInvocationServiceImpl.shutdown(ClientSmartInvocationServiceImpl.java:30)
	at com.hazelcast.client.impl.HazelcastClientInstanceImpl.doShutdown(HazelcastClientInstanceImpl.java:590)
	at com.hazelcast.client.impl.LifecycleServiceImpl.shutdown(LifecycleServiceImpl.java:108)
	at com.hazelcast.client.impl.LifecycleServiceImpl.terminate(LifecycleServiceImpl.java:114)
	at com.hazelcast.client.test.TestHazelcastFactory.terminateAll(TestHazelcastFactory.java:126)
	at com.hazelcast.client.ClientRegressionWithMockNetworkTest.cleanup(ClientRegressionWithMockNetworkTest.java:80)
```

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.maintenance-IbmJDK1.8/com.hazelcast$hazelcast-client/60/testReport/junit/com.hazelcast.client/ClientRegressionWithMockNetworkTest/testClientConnectionEvents/

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x-IbmJDK1.7/com.hazelcast$hazelcast-client/861/testReport/junit/com.hazelcast.client/ClientRegressionWithMockNetworkTest/testClientConnectionEvents/

```
java.lang.AssertionError: null
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at com.hazelcast.client.ClientRegressionWithMockNetworkTest.testClientConnectionEvents(ClientRegressionWithMockNetworkTest.java:270)
```

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.maintenance-OracleJDK1.6/com.hazelcast$hazelcast-client/366/testReport/junit/com.hazelcast.client/ClientRegressionWithMockNetworkTest/testClientConnectionEvents/

```
java.lang.AssertionError: null
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at com.hazelcast.client.ClientRegressionWithMockNetworkTest.testClientConnectionEvents(ClientRegressionWithMockNetworkTest.java:272)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast%20Maintenance/job/Hazelcast-3.maintenance/com.hazelcast$hazelcast-client/1302/testReport/junit/com.hazelcast.client/ClientRegressionWithMockNetworkTest/testClientConnectionEvents/

```
java.lang.AssertionError: null
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at com.hazelcast.client.ClientRegressionWithMockNetworkTest.testClientConnectionEvents(ClientRegressionWithMockNetworkTest.java:272)
```

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.maintenance/com.hazelcast$hazelcast-client/1302/testReport/junit/com.hazelcast.client/ClientRegressionWithMockNetworkTest/testClientConnectionEvents/

```
java.lang.AssertionError: null
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at com.hazelcast.client.ClientRegressionWithMockNetworkTest.testClientConnectionEvents(ClientRegressionWithMockNetworkTest.java:272)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast%20Maintenance/job/Hazelcast-3.maintenance-OracleJDK1.6/com.hazelcast$hazelcast-client/383/testReport/junit/com.hazelcast.client/ClientRegressionWithMockNetworkTest/testClientConnectionEvents/

```
java.lang.AssertionError: null
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at com.hazelcast.client.ClientRegressionWithMockNetworkTest.testClientConnectionEvents(ClientRegressionWithMockNetworkTest.java:272)
```

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.maintenance-ZuluJDK7/com.hazelcast$hazelcast-client/92/testReport/junit/com.hazelcast.client/ClientRegressionWithMockNetworkTest/testClientConnectionEvents/

**master**
```
java.lang.AssertionError: null
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at com.hazelcast.client.ClientRegressionWithMockNetworkTest.testClientConnectionEvents(ClientRegressionWithMockNetworkTest.java:269)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast%20Maintenance/job/Hazelcast-3.maintenance-IbmJDK1.7/com.hazelcast$hazelcast-client/409/testReport/junit/com.hazelcast.client/ClientRegressionWithMockNetworkTest/testClientConnectionEvents/

**maintenance**

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast%20Maintenance/job/Hazelcast-3.maintenance-IbmJDK1.7/com.hazelcast$hazelcast-client/409/testReport/junit/com.hazelcast.client/ClientRegressionWithMockNetworkTest/testClientConnectionEvents/

--- a/hazelcast-client/src/test/java/com/hazelcast/client/ClientRegressionWithMockNetworkTest.java
+++ b/hazelcast-client/src/test/java/com/hazelcast/client/ClientRegressionWithMockNetworkTest.java
@@ -261,11 +261,11 @@ public class ClientRegressionWithMockNetworkTest extends HazelcastTestSupport {
         clientConfig.getNetworkConfig().setConnectionAttemptLimit(100);
         hazelcastFactory.newHazelcastClient(clientConfig);
 
-        instance.shutdown();
+        hazelcastFactory.shutdownAllMembers();
 
         hazelcastFactory.newHazelcastInstance();
 
-        assertTrue(latch.await(60, TimeUnit.SECONDS));
+        assertTrue("LifecycleState failed" , latch.await(60, TimeUnit.SECONDS));
     }
 
 

--- a/hazelcast-client/src/test/java/com/hazelcast/client/test/TestHazelcastFactory.java
+++ b/hazelcast-client/src/test/java/com/hazelcast/client/test/TestHazelcastFactory.java
@@ -116,6 +116,10 @@ public class TestHazelcastFactory extends TestHazelcastInstanceFactory {
         super.shutdownAll();
     }
 
+    public void shutdownAllMembers() {
+        super.shutdownAll();
+    }
+
     @Override
     public void terminateAll() {
         if (!mockNetwork) {

--- a/hazelcast/src/test/java/com/hazelcast/test/TestHazelcastInstanceFactory.java
+++ b/hazelcast/src/test/java/com/hazelcast/test/TestHazelcastInstanceFactory.java
@@ -200,6 +200,7 @@ public class TestHazelcastInstanceFactory {
     public void shutdownAll() {
         if (mockNetwork) {
             registry.shutdown();
+            addresses.clear();
         } else {
             Hazelcast.shutdownAll();
         }

